containers/src/Kokkos_OffsetView.hpp

```diff
@@ -928,78 +928,97 @@ class OffsetView : public ViewTraits<DataType, Properties...> {
  private:
   enum class subtraction_failure {
     none,
-    non_negative,
+    negative,
     overflow,
   };
 
-  // Subtraction should return a strictly positive number and not overflow
-  template <typename I>
-  KOKKOS_INLINE_FUNCTION static subtraction_failure check_subtraction(I lhs,
-                                                                      I rhs) {
-    if (lhs <= rhs) return subtraction_failure::non_negative;
-
-    if (std::is_signed<I>::value) {
-      using U = typename std::make_unsigned<I>::type;
-      U diff  = static_cast<U>(lhs) - static_cast<U>(rhs);
-      U max   = (static_cast<U>(0) - static_cast<U>(1)) >> 1;
-      if (diff > max) return subtraction_failure::overflow;
-    }
+  // Subtraction should return a non-negative number and not overflow
+  KOKKOS_INLINE_FUNCTION static subtraction_failure check_subtraction(
+      int64_t lhs, int64_t rhs) {
+    if (lhs < rhs) return subtraction_failure::negative;
+
+    if (static_cast<uint64_t>(-1) / static_cast<uint64_t>(2) <
+        static_cast<uint64_t>(lhs) - static_cast<uint64_t>(rhs))
```

Comment:
I get how you compute `max` which is the largest integer that can be represented.  I just don't get how it works for `diff` if any of the two "operands" is negative.

Reply:
Nevermind I get it now :)